### PR TITLE
Add --pull to docker build command

### PR DESCRIPTION
**Pull request by MichaelSimons**

### CODE REVIEW

**Comment by MichaelSimons** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -14,7 +14,7 @@ platformList.each { platform ->
                 batchFile("powershell -NoProfile -File .\\Microsoft.DotNet.ImageBuilder\\build.ps1 -CleanupDocker")
             }
             else {
-                shell("docker build -t runner -f ./Microsoft.DotNet.ImageBuilder/Dockerfile.linux.runner .")
+                shell("docker build -t runner -f ./Microsoft.DotNet.ImageBuilder/Dockerfile.linux.runner --pull .")
```
Comment (ravimeda) [inline netci.groovy]: Consider putting options before the path -
`docker build -t runner --pull -f ./Microsoft.DotNet.ImageBuilder/Dockerfile.linux.runner .`

Comment (MichaelSimons) [reply netci.groovy]: What is the value in doing this?  `-t` and `-f` and the more common args.

Comment (ravimeda) [reply netci.groovy]: Prefer going by the help text - `Usage:  docker build [OPTIONS] PATH | URL | -`

Comment (MichaelSimons) [reply netci.groovy]: But that isn't the order you proposed is it?  Your suggestion was the reverse order.  I personally think the tag should be listed first as it is the one identifier you need to know in order to use the resulting image.  Leaving as is.

**Pull request merged**